daphne/src/testing.rs

```diff
@@ -61,23 +67,31 @@ pub const COLLECTOR_BEARER_TOKEN: &str = "syfRfvcvNFF5MJk4Y-B7xjRIqD_iNzhaaEB9mY
 pub(crate) struct MockAggregator {
     tasks: HashMap<Id, DapTaskConfig>,
     hpke_receiver_config_list: Vec<HpkeReceiverConfig>,
+    report_store: ReportStore,
```

Comment:
Thinking ahead, how does the leader pick a set of reports to aggregate for a given task? It'll have scan through this monololthic list to find a set of reports with matching task IDs. This should be something like an `HashMap<Id, Vec<Report>>`

Reply:
I have implemented a struct called `BatchInfo` that houses the three data used to store `ReportStore` in DO: `task_id`, `window`, and `bucket`.
I thought some compatibility with `DaphneWorker` would be good, but if you think this is overkill, I can remove `BatchInfo` and replace it with task ID.

Reply:
`BatchInfo` is now renamed into `BucketInfo` to better represent what it is.
The latest commit also removes `bucket`, which is only necessary for DOs.